src/rshapes.c

```diff
@@ -1734,6 +1736,23 @@ bool CheckCollisionPointLine(Vector2 point, Vector2 p1, Vector2 p2, int threshol
     return collision;
 }
 
+bool CheckCollisionRayLine(Vector2 rayPosition, Vector2 rayDirection, Vector2 startPos, Vector2 endPos, Vector2 *collisionPoint)
+{    
+    bool collision = false;
+    Vector2 offsetPosition;
+    offsetPosition.x = rayPosition.x + rayDirection.x;
+    offsetPosition.y = rayPosition.y + rayDirection.y;
```

Comment:
This is never used as a vector, always as separate values, it may be best to skip the allocation of a vector and just use x and y values directly like the other collisions do.

Reply:
That's fair. This was adapted from an iteration that used raymath and vector operations, so there are some remnants.

Reply:
I would look over the other collision code and see what you can do to match the style.

Reply:
Good point, doing that rn.